crates/oxc_minifier/src/compressor/fold.rs

```diff
@@ -180,4 +194,265 @@ impl<'a> Compressor<'a> {
 
         None
     }
+
+    fn try_fold_unary_operator<'b>(
+        &mut self,
+        unary_expr: &'b mut UnaryExpression<'a>,
+    ) -> Option<Expression<'a>> {
+        // TODO: I want compress children first, so we can fold expression like `- - 4`
+        // But our ast will visit more than once, is there better way?
+        self.fold_expression(&mut unary_expr.argument);
```

Comment:
I want to compress children first, so we can fold expressions like `- - 4`, But our AST will visit more than once, is there a better way?

Reply:
Interesting, let's figure this out later.